code/game/objects/items/tcg/tcg.dm

```diff
@@ -71,6 +71,121 @@ GLOBAL_LIST_EMPTY(cached_cards)
 /obj/item/tcgcard/proc/zoom_out()
 	transform = matrix(0.3,0,0,0,0.3,0)
 
+/obj/item/tcgcard/update_icon_state()
+	. = ..()
+	if(!flipped)
+		var/datum/card/template = GLOB.cached_cards[series]["ALL"][id]
+		name = template.name
+		desc = template.desc
+		icon_state = template.icon_state
+
+	else
+		name = "Trading Card"
+		desc = "It's the back of a trading card... no peeking!"
+		icon_state = "cardback"
+
+/obj/item/tcgcard/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/tcgcard))
+		var/obj/item/tcgcard/second_card = I
+		var/obj/item/tcgcard_deck/new_deck = new /obj/item/tcgcard_deck(drop_location())
+		new_deck.flipped = flipped
+		user.transferItemToLoc(second_card, new_deck)//Start a new pile with both cards, in the order of card placement.
+		user.transferItemToLoc(src, new_deck)
+		new_deck.update_icon_state()
+	return ..()
+
+/**
+  * A stack item that's not actually a stack because ORDER MATTERS with a deck of cards!
+  * The "top" card of the deck will always be the bottom card in the stack for our purposes.
+  */
+/obj/item/tcgcard_deck
+	name = "Trading Card Pile"
+	desc = "A stack of TCG cards."
+	icon = DEFAULT_TCG_DMI_ICON
+	icon_state = "deck_up"
+	obj_flags = UNIQUE_RENAME
+	var/flipped = FALSE
+	var/static/radial_draw = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_draw")
+	var/static/radial_shuffle = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_shuffle")
+	var/static/radial_pickup = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_pickup")
+
+/obj/item/tcgcard_deck/Initialize()
+	. = ..()
+	LoadComponent(/datum/component/storage/concrete/tcg)
+
+/obj/item/tcgcard_deck/update_icon_state()
+	. = ..()
+	if(flipped)
+		icon_state = "deck_down"
+	else
+		icon_state = "deck_up"
+
+/obj/item/tcgcard_deck/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>\The [src] has [contents.len] cards inside.</span>"
+
+/obj/item/tcgcard_deck/attack_hand(mob/user)
+	var/list/choices = list(
+		"Draw" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_draw"),
+		"Shuffle" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_shuffle"),
+		"Pickup" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_pickup"),
+		"Flip" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_flip"),
+		)
+	var/choice = show_radial_menu(user, src, choices, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
+		return
+	switch(choice)
+		if("Draw")
+			draw_card(user)
+		if("Shuffle")
+			shuffle_deck(user)
+		if("Pickup")
+			user.put_in_hands(src)
+		if("Flip")
+			flipped = !flipped
+			contents = reverseRange(contents)
```

Comment:
Similar to the other comment, this one also seems to have an issue when being called on the contents of the object.

Reply:
![image](https://user-images.githubusercontent.com/58055496/91147838-aeff1d80-e66d-11ea-8aa9-ed055ca317c5.png)
aight so fun fact, shuffle(), unlike reverseRange(), Operates on a Copy() of the list that gets passed to it, which allows it to bypass the fucking restriction on contents. you just need to throw a .Copy() on the end so reverseRange gets a mutable proc for it to play with.